R/dose_grid_search.R

```diff
@@ -279,6 +270,22 @@ simulate_dose_interval <- function(value,
     )
   }
 
+  ## Calculate time at which target needs to be checked
+  ## This is constant for dose-optimization, but differs for interval-based
+  ## optimization.
+  target_time <- get_sampling_times_from_scheme(
+    target_design$scheme, reg
+  )
+  if(length(target_time) > 1 && !target_design$type %in% c("auc", target_types_time)) {
+    target_time <- target_time[1]
+  }
+  if(target_design$type %in% c("auc", target_types_time)) {
+    if(length(target_time) != 2) {
+      stop("Need start and end of observation interval as vector target_time.")
+    }
+  }
+  t_obs <- target_time
```

Comment:
the code above is obviously fine but maybe it makes more sense to name the `target_time` variable `t_obs` for this context. its current name is a relic of the code's former location.